app/src/main/java/ch/epfl/sweng/favors/location/FakeGeocoder.java

```diff
@@ -0,0 +1,38 @@
+package ch.epfl.sweng.favors.location;
+
+import android.location.Address;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+public class FakeGeocoder extends GeocoderDispatcher {
+    public static final String FAKE_LOCATION_CITY = "Fake Lausanne";
+    public static final double FAKE_LATITUDE = 46.5196535;
+    public static final double FAKE_LONGITUDE = 6.6322734;
+
+    private ArrayList<Address> addresses;
+
+    private FakeGeocoder(){
+        addresses = new ArrayList<>();
+        addresses.add(new Address(Locale.getDefault()));
+        addresses.get(0).setLocality(FAKE_LOCATION_CITY);
+        addresses.get(0).setLatitude(FAKE_LATITUDE);
+        addresses.get(0).setLongitude(FAKE_LONGITUDE);
+    }
+
+    public static FakeGeocoder getGeocoder() {
+        return new FakeGeocoder();
```

Comment:
I prefered to use a factory than a singleton pattern. I made this choice because I do not see the need to have only one Geocoder since Geocoder is not a singleton in the android API.